src/actions.js

```diff
@@ -143,6 +144,19 @@ const actions = {
       mode,
     }
   },
+  addLanguage(languageId, evaluate, displayName, codeMirrorName, keybinding) {
+    return {
+      type: 'ADD_LANGUAGE',
+      languageId,
+      languageDefinition: {
+        languageId,
+        evaluate,
+        displayName,
+        codeMirrorName,
```

Comment:
Maybe this should be called `syntaxHight` so as to semantically decouple it specifically from CodeMirror, yeah? I know this is kind of tough because let's say we change editors and we have to change the `codeMirrorName` value to some other string. At least we don't have to update all the symbols I guess.

Reply:
I see your point.  I think it's probably ok to go with the generic variable name (`syntaxHighlight`) and worry about the possibility of maybe needing to rename what languages are called by the syntax highlighting tool later.

Reply:
@mdboom I ported this over from your code without thinking about the names much, but as i reflect on it now for a second I actually think your first intuition on the naming conventino was the right one -- it seems that using `codeMirrorName` is actually maybe semantically useful here. I'm not entirely sure how the (React) CodeMirror plugin situation works, but it looks like there are a few modules that modify the CodeMirror module upon import, and that they define magic keywords in CodeMirror that are specific to CodeMirror. So using `codeMirrorName` is kind of a red flag/reminder that if you are working on this stuff, looking through the Iodide code for our language naming conventions won't be enough, you'll have to look at the CodeMirror docs too in all likelihood.

Reply:
but maybe `codeMirrorMode` is better:
https://codemirror.net/doc/manual.html#option_mode
https://codemirror.net/mode/index.html

i think these are the CodeMirror-specific mode names that we'll need, and that the other plugins (autocomplete, comment, etc) key off of.